.github/workflows/scripts/upload-demo-images.sh

```
@@ -0,0 +1,239 @@
+    
+    # Get the user's actual commit SHA (from the PR branch)
+    # Use the commit_sha passed in, which is the actual PR head commit
+    local user_commit_sha="$commit_sha"
+    
+    # FIRST: Backup all generated demos BEFORE switching branches
+    log_step "Backing up generated demos before branch switch..."
+    local backup_dir
+    backup_dir=$(mktemp -d)
+    
+    # Set up cleanup trap
+    trap "rm -rf '$backup_dir'" EXIT
```

Shellcheck (0.10.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)